rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -115,13 +128,21 @@ impl BaseMetadataBuilder {
 
     pub async fn get_proof(&self, nonce: u32, checkpoint: Checkpoint) -> Result<Option<Proof>> {
         const CTX: &str = "When fetching message proof";
-        let proof = self
-            .origin_prover_sync
-            .read()
-            .await
-            .get_proof(nonce, checkpoint.index)
-            .context(CTX)?;
-
+        let proof = retry(constant_backoff(), || async {
```

Comment:
I don't think we should need to retry here because the `get_proof` that's called returns an Option that we can propagate just propagate up. Then we can rely on the retry logic we get from our message submitter when metadata building is None. This assumes the aggregation ISM logic is robust enough where if we have a 1/2 setup of the merkle proof & message ID ISMs, if merkle proof metadata building returns None but the message id based metadata building returns Some, then we're still able to process things correctly

I think this is actually a bit dangerous to have a retry here because it's possible for someone to send a message without inserting it into the merkle tree on the origin, but on the destination they can configure an ISM that requires a merkle proof. In that case then the relayer will never be able to construct a merkle proof, so it'd be retrying unnecessarily

(maybe i'm misreading this tho, by my read this is retrying getting the proof if `get_proof` returns none?)

Reply:
That makes sense. So we should be able to remove the `backoff` dep entirely